File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums) - 1
        while left <= right:
            middle = (left + right) // 2
            if nums[middle] == target:
                return middle
            elif nums[left] <= target <= nums[middle] or nums[middle] < nums[left] <= target or target < nums[middle] < nums[left]:
                right = middle - 1
            else:
                left = middle + 1
        return -1

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        size_t len = nums.size();
        for (size_t i = 0; i < len; i++) {
            for (size_t j = i+1; j < len; j++) {
                if (nums[i] + nums[j] == target) {
                    return vector<int>{i, j};
                }
            }
        }
    }
};
